Close #251: Implement seed screen

valentinewallace opened the pull request:
Hey, I was spending way too long figuring out how to get the SplitBackground to scale with the height of the content so I made a compromise and made the bottom half scrollable. LMK if you have any tips on that :)

tanx commented on src/component/button.js:
```diff
@@ -78,7 +78,13 @@ const glasStyles = StyleSheet.create({
+  disabled,
+  children,
+  style,
+  textStyle,
```
We should avoid setting two style objects in one component. If necessary, I'd prefer pulling the inner components out to make them styleable. But it actually shouldn't be necessary to style the text size for this button.

tanx replied:
In general the fonts are pretty big. We should try to fit the design into the default viewport size https://github.com/lightninglabs/lightning-app/blob/dd046e284bf13b8c09653eca5c29f5bcb8039d05/public/electron.js#L73-L74

This is what that currently looks like...
![image](https://user-images.githubusercontent.com/1374174/41066755-567da66c-69e3-11e8-8020-ee970742e8bf.png)

tanx commented on src/view/seed.js:
```diff
@@ -0,0 +1,111 @@
+    alignSelf: 'center',
+    textAlign: 'center',
+    fontSize: font.sizeL + 2 * 2,
+    lineHeight: font.lineHeightL + 2 * 4,
```
If you need to multiply the fonts by a constant factor, that's probably a good sign that we need a new font size. Considering how large the seed view renders on the default viewport size though, I'm pretty sure we can do with the font sizes we currently already have.

tanx commented on src/component/list.js:
```diff
@@ -55,6 +55,9 @@ export class List extends Component {
         renderHeader={this.props.renderHeader}
         renderRow={this.props.renderItem}
         enableEmptySections={true}
+        horizontal={this.props.horizontal}
```
Why do we want the list to scroll horizontally?

valentinewallace replied:
True, I forget why i put that in. took out

tanx commented:
I added some minor changes in the two commits.

tanx commented:
I added some minor changes (see my commits).

> Hey, I was spending way too long figuring out how to get the SplitBackground to scale with the height of the content so I made a compromise and made the bottom half scrollable. LMK if you have any tips on that :)

I changed it so the the top container is a flex View instead of having two MainContent components in one screen, which doesn't really make much sense semantically.

tanx commented on src/view/seed.js:
```diff
@@ -0,0 +1,99 @@
+    justifyContent: 'center',
+    alignItems: 'center',
+    paddingTop: 25,
+    paddingBottom: 25,
```
I'm wondering if it's possible to have the List float in the center of the MainView instead of adjusting the position via paddingTop?

tanx commented on stories/screen.js:
```diff
@@ -102,7 +103,8 @@ storiesOf('Screens', module)
   ))
-  .add('Loader - First Time', () => <Loader />);
+  .add('Loader - First Time', () => <Loader />)
+  .add('Seed', () => <Seed seedMnemonic={store.seedMnemonic} />);
```
We should stay consistent and always inject the complete here.

tanx commented on src/view/seed.js:
```diff
@@ -0,0 +1,99 @@
+        )}
+        contentStyle={styles.wordList}
+        initialListSize={24}
+      />
```
I'm also wondering whether a ListView is the right component here. Seems like it would be easier to just render the 24 boxes in a forEach loop inside a View. Then we wouldn't have to style the ListView either.

tanx commented on src/view/seed.js:
```diff
@@ -0,0 +1,124 @@
+      <WordList seedMnemonic={store.seedMnemonic.slice()} />
+      <GlasButton onPress={() => {}}>Next</GlasButton>
+    </MainContent>
+  </SplitBackground>
```
I split up the view into components.

valentinewallace replied:
Ahh, good one

tanx commented on src/view/seed.js:
```diff
@@ -0,0 +1,124 @@
+  copyTxt: {
+    textAlign: 'center',
+    maxWidth: 410,
+  },
```
I also cleaned up the font sizes to be in line with the rest of the instances.

tanx commented on src/view/seed.js:
```diff
@@ -0,0 +1,124 @@
+        <Word word={word} index={i + 1} key={i} />
+      ))}
+    </View>
+  </View>
```
This wasn't rendering correctly. So I changed the styled. Also `key` is a reserved attribute and can't be used as a prop...

![image](https://user-images.githubusercontent.com/1374174/41222989-d2c8c408-6d68-11e8-9584-fbd35d2db669.png)

valentinewallace replied:
Hmm, that's weird, i was getting a console error saying `key` was required when you're using a `for` loop like that... i'll re-run it

valentinewallace commented:
Great feedback, thanks. I rebased onto master and added a minor `title` style change to prevent this edge case: 
![first_write_down2](https://user-images.githubusercontent.com/10647624/41259311-06614668-6d88-11e8-85d6-0ea0beef3d8b.png)

tanx commented:
> Great feedback, thanks. I rebased onto master and added a minor title style change to prevent this edge case:

@valentinewallace good catch. I think this is what you meant https://github.com/lightninglabs/lightning-app/commit/7d2c55c97619cc1a6454f2d643b84a5491a1f8a1 ?  If you have changes/comments we can keep iterating on the branch next time until it's ready to merge.

Other than that great to finally get this merged. Good job 👍
